contracts/cw-ics721-bridge/src/contract.rs

```diff
@@ -199,17 +169,17 @@ fn do_receive_nft(
     let UniversalNftInfoResponse { token_uri, .. } = deps.querier.query_wasm_smart(
         info.sender,
         &cw721::Cw721QueryMsg::NftInfo {
-            token_id: token_id.clone(),
+            token_id: token_id.clone().into(),
         },
     )?;
 
     let ibc_message = NonFungibleTokenPacketData {
         class_id: class_id.clone(),
         class_uri,
+        class_data: None, // TODO: check for values we need to forward from a source chain.
+        token_data: None,
         token_ids: vec![token_id.clone()],
-        token_uris: vec![token_uri.unwrap_or_default()], /* Currently token_uri is optional in
```

Comment:
Additional Q: When we actually start unwrapping fields such as royalty, 1) where would that unwrapping occur and 2) when would that data get packed?

Reply:
it'll happen in the `Mint` callback message. there, we'd try to parse the `tokenData` base64 encoded string into royalty JSON. in these changes, i pass `tokenData` through to `Mint` even though we don't use it to make adding this in the future simple.